### PR TITLE
[TF API] Implement DoubleValueDataset and and random dataset ops

### PR DESCRIPTION
**Pull request by rxwei**
- Add a `DoubleValueDataset` to represent a dataset whose elements are tuples of tensors.

- Add random initializer `init(randomSeed:)`.

- Add `shuffled(sampleCount:randomSeed:)` and `batched(_:)`.

**Note**: Arbitrary arity `Dataset` types are not possible until two new features of `#tfop` are implemented. Please refer to [SR-8573](https://bugs.swift.org/browse/SR-8573) and [SR-8574](https://bugs.swift.org/browse/SR-8574) for more details.

Resolves [SR-8597](https://bugs.swift.org/browse/SR-8597).

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci  please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

File: stdlib/public/TensorFlow/Dataset.swift [hunk 1]
```diff
@@ -139,3 +192,197 @@ extension SingleValueDatasetIterator : IteratorProtocol {
                                 output_shapes: [elementShape]))
   }
 }
+
+//===----------------------------------------------------------------------===//
+// Double value dataset
```
Comment (marcrasi) [inline stdlib/public/TensorFlow/Dataset.swift]: Is this basically a copy-and-paste of the single value dataset, with some types changed?

Add a comment above SingleValueDataset and a comment above DoubleValueDataset describing this relationship and reminding people to keep them in sync. Also explain why they're duplicates, like you do in the PR description.

**Review by marcrasi — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux Platform

**Comment by rxwei** [comment]
@swift-ci  please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
Thanks for the review!

**Pull request merged**